go/apis/kube-mail/v1alpha1/smtpserver_types.go

```diff
@@ -11,14 +11,22 @@ const (
 	SMTPAuthTypeSCRAMSHA1 SMTPAuthType = "SCRAM-SHA-1"
 )
 
+type SMTPConnectType string
+
+const (
+	SMTPConnectTypePlain     SMTPConnectType = "plain"
+	SMTPConnectTypeSSL     SMTPConnectType = "ssl"
+	SMTPConnectTypeSTARTTLS SMTPConnectType = "starttls"
+)
+
 type SMTPServerSpec struct {
 	Server string `json:"server"`
 
 	// +optional
 	Port int `json:"port,omitempty"`
 
-	// +optional
-	TLS bool `json:"tls,omitempty"`
+	// +kubebuilder:validation:Enum=plain;ssl;starttls
+	Connect SMTPConnectType `json:"connect" ts_type:"connect"`
```

Comment:
I would indeed prefer keeping backwards-compatibility here. In this case, this would include keeping the `TLS bool` property (and, in the same course, denoting it as `deprecated`), and adding `Connect` as an additional property (which would also need to be optional.

```suggestion
	// DEPRECATED: Use Connect, instead
	// +optional
	TLS bool `json:"tls,omitempty"`
	
	// +kubebuilder:validation:Enum=plain;ssl;starttls
	// +optional
	Connect SMTPConnectType `json:"connect,omitempty" ts_type:"connect"`
```

Reply:
@martin-helmich I've amended my commit to try and provide a solution for this. Let me know what you think.